tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -1597,6 +1597,39 @@ public:
       return definedColumns;
    }
 
+   /// \brief Checks if a column is present in the dataset
+   /// \return true if the column is available, false otherwise
+   ///
+   /// This method checks if a column is part of the input ROOT dataset, has
+   /// been defined or can be provided by the data source.
+   ///
+   /// Example usage:
+   /// ~~~{.cpp}
+   /// ROOT::RDataFrame base(1);
+   /// auto rdf = base.Define("definedColumn", [](){return 0;});
+   /// rdf.HasColumn("definedColumn"); // true: we defined it
+   /// rdf.HasColumn("rdfentry_"); // true: it's always there
+   /// rdf.HasColumn("foo"); // false: it is not there
+   /// ~~~
+   bool HasColumn(std::string_view columnName)
+   {
+      if(fCustomColumns.HasName(columnName)) return true;
+
+      if (auto tree = fLoopManager->GetTree()) {
+         const auto branchNames = RDFInternal::GetBranchNames(*tree, /*allowDuplicates=*/false);
```

Comment:
shouldn't `allowDuplicates` be `true` to let users search a given column in all different ways it can be called?

Reply:
good point.